lib/src/settings.rs

```diff
@@ -159,7 +159,9 @@ impl UserSettings {
             // For compatibility with old config files (<0.8.0)
             self.config
                 .get_string("ui.default-revset")
-                .unwrap_or_else(|_| "@ | ancestors((remote_branches() | tags()).., 2)".to_string())
+                .unwrap_or_else(|_| {
+                    "@ | ancestors(immutable_heads().., 2) | heads(::immutable_heads())".to_string()
+                })
```

Comment:
`heads(::immutable_heads())` can be rephrased as `heads(immutable_heads())`?

btw, `heads()` can be expensive if we have old tags, so `| trunk()` might be better approximation.

Reply:
Good point.